dolphin/api/v1/storages.py

```diff
@@ -42,11 +47,64 @@ def validate_parameters(data, required_parameters,
             msg = _("Required parameter %s is empty.") % parameter
             raise exc_response(explanation=msg)
 
+LOG = log.getLogger(__name__)
+
+
+def build_storages(storages):
+    # Build list of storages
+    views = [build_storage(storage)
+             for storage in storages]
+    return dict(storages=views)
+
+
+def build_storage(storage):
+    view = copy.deepcopy(storage)
+    return view
+
 
 class StorageController(wsgi.Controller):
 
     def index(self, req):
-        return dict(name="Storage 1")
+
+        supported_filters = [
+            'name',
+            'vendor',
+            'model',
+            'status',
+        ]
+        filters = {}
+        filters.update(req.GET)
+        # update options  other than filters
+        if 'sort_key' in filters:
+            sort_key = [filters['sort_key']]
+        else:
+            sort_key = None
+        if 'sort_dir' in filters:
+            sort_dir = [filters['sort_dir']]
+        else:
+            sort_dir = None
+        if 'limit' in filters:
+            limit = filters['limit']
+        else:
+            limit = None
+        if 'offset' in filters:
+            offset = filters['offset']
+        else:
+            offset = None
+        # strip out options except supported filter options
```

Comment:
Validation checking for all input  parameters may be required. And as sort_key, sort_dir, limit, marker, offset are common parameters, it's better to build common method to do those checking.

Reply:
optimizeed the checking part.. plz check

Reply:
For those common parameters that many controllers will use, is it better to create common method to do checking, so other controllers can reuse it?

Reply:
will make  this as a common function based the next API implemnetaion..